test/unit/horizon_path_test.js

```diff
@@ -0,0 +1,241 @@
+    StellarSdk.Network.useTestNetwork();
+    let keypair = StellarSdk.Keypair.random();
+    let account = new StellarSdk.Account(keypair.publicKey(), "56199647068161");
+    this.transaction = new StellarSdk.TransactionBuilder(account)
+      .addOperation(StellarSdk.Operation.payment({
+        destination: "GASOCNHNNLYFNMDJYQ3XFMI7BYHIOCFW3GJEOWRPEGK2TDPGTG2E5EDW",
+        asset: StellarSdk.Asset.native(),
+        amount: "100.50"
+      }))
+      .build();
+    this.transaction.sign(keypair)
+
+  });
+
+  afterEach(function () {
+    this.axiosMock.verify();
+    this.axiosMock.restore();
+  });
+
+  describe('server url without folder path: https://acme.com:1337', function () {
+    let server = new StellarSdk.Server('https://acme.com:1337');
+
+    let callbuilder = null;
+
+
+
+    it("server.accounts()", function () {
+      callbuilder = server.accounts();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/accounts");
+    });
+
+
+    it('server.accounts().accountId("fooAccountId")', function () {
+      callbuilder = server.accounts().accountId("fooAccountId");
+      callbuilder.checkFilter();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/accounts/fooAccountId");
+    });
+
+
+    it('server.transactions()', function () {
+      callbuilder = server.transactions();
+      callbuilder.checkFilter();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/transactions");
+    });
+
+
+    it('server.transactions().transaction("fooTransactionId")', function () {
+      callbuilder = server.transactions().transaction("fooTransactionId");
+      callbuilder.checkFilter();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/transactions/fooTransactionId");
+
+    });
+
+
+
+    it('server.transactions().transaction("fooTransactionId")', function () {
+      callbuilder = server.transactions().transaction("fooTransactionId");
+      callbuilder.checkFilter();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/transactions/fooTransactionId");
+
+    });
+
+
+
+    it('server.transactions().forAccount("fooAccountId")', function () {
+      callbuilder = server.transactions().forAccount("fooAccountId");
+      callbuilder.checkFilter();
+      expect(callbuilder.url.toString()).to.be.equal("https://acme.com:1337/accounts/fooAccountId/transactions");
+
+    });
+
+    it('server.submitTransaction', function (done) {
+
+      this.axiosMock.expects('post')
+        .withArgs(sinon.match('https://acme.com:1337/transactions', 'bogus'))
+        .returns(Promise.resolve({
+          ok: true
+        }));
+
+      server.submitTransaction(this.transaction)
+        .then(response => {
+          done();
+        })
+        .catch(function (err) {
+          done(err);
+        });
+
+    });
+
+
+
+
+  });
+
+
+  describe('server url with folder path: https://acme.com:1337/folder/', function () {
```

Comment:
A lot of code duplication here. Could you just create an array of server URLs and then create a loop that iterates over them and runs tests? This will make this file size 1/3 of the current size.